proxy/src/coco/git_helper.rs

```diff
@@ -25,22 +24,117 @@ fn is_executable(path: &std::path::PathBuf) -> Result<bool, error::Error> {
 ///   * Could not get the current working directory.
 ///   * Could not create the path to binary directory.
 ///   * Could not copy helper executable to the binary directory.
-pub fn setup() -> Result<(), error::Error> {
+pub fn setup(src_dir: &path::PathBuf, dst_dir: &path::PathBuf) -> Result<(), error::Error> {
     log::info!("Making sure git-remote-rad helper is set up");
 
-    let bin_dir = config::bin_dir()?;
-    let full_dest_path = bin_dir.join(GIT_REMOTE_RAD);
+    let helper_bin_src = src_dir.join(GIT_REMOTE_RAD);
+    let helper_bin_dst = dst_dir.join(GIT_REMOTE_RAD);
 
-    if full_dest_path.exists() && is_executable(&full_dest_path)? {
-        log::debug!("Git helper already exists at: {:?}", full_dest_path);
+    if helper_bin_dst.exists() && is_executable(&helper_bin_dst)? {
+        log::debug!("Git helper already exists at: {:?}", helper_bin_dst);
         return Ok(());
     }
 
-    let proxy_path = config::proxy_path()?;
-
-    fs::create_dir_all(bin_dir)?;
-    fs::copy(proxy_path.join(GIT_REMOTE_RAD), full_dest_path.clone())?;
-    log::debug!("Copied git-remote-rad helper to: {:?}", full_dest_path);
+    fs::create_dir_all(dst_dir)?;
+    fs::copy(helper_bin_src, helper_bin_dst.clone())?;
+    log::debug!("Copied git-remote-rad helper to: {:?}", helper_bin_dst);
 
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use crate::coco;
+    use crate::error;
+    use std::fs;
+    use std::os::unix::fs::PermissionsExt;
+
+    // when the ~/.radicle/bin directory doesn't exist at all
+    #[tokio::test]
+    async fn setup_creates_destination_directory_if_none_exists() -> Result<(), error::Error> {
+        let tmp_src_dir = tempfile::tempdir().expect("failed to create source tempdir");
+        let src_git_helper_bin_path = tmp_src_dir.path().join(coco::git_helper::GIT_REMOTE_RAD);
```

Comment:
fyi: you can use `super` :)

Reply:
Good point!

https://github.com/radicle-dev/radicle-upstream/pull/769/commits/9944006e19da6edde1c0754b73a1e364310b5568